src/aiida_quantumespresso/workflows/pw/base.py

```diff
@@ -180,6 +180,10 @@ def get_builder_from_protocol(
         parameters['SYSTEM']['ecutwfc'] = cutoff_wfc
         parameters['SYSTEM']['ecutrho'] = cutoff_rho
 
+        #If 2D periodicity set assume_isolate
+        if (True, True, False) == structure.pbc:
+            parameters['SYSTEM']['assume_isolated'] = '2D'
```

Comment:
I see, so this correction can only be used when the first two lattice vectors are entirely in the x-y plane. So running with `(False, True, True)` should not be allowed for this reason? Should the third lattice vector be perpendicular to this plane?

Reply:
@mbercx Thank you for the question. So I included this option given the suggestion from @demiranda-gabriel , Usually If i were to do a slab simulation i would omit this part and just be sure that my structure has enough vacuum in the perpendicular direction of the slab. According to the manual this option is only for xy-plane. Though I am pro removing this line.

Reply:
Hmm, since the `get_builder_from_protocol` method is meant to provide a set of sensible defaults, it might be reasonable to keep it since the user can always override it. One sentence in the QE documentation still worries me:

```
Total energy, forces and stresses are computed in a two-dimensional framework.
```

@demiranda-gabriel So QE would not consider any z-components of the forces? Wouldn't we want to include these when e.g. optimizing the surface layers?

Reply:
Nevermind, it clearly does. The initial forces and stresses:

```console
     Forces acting on atoms (cartesian axes, Ry/au):

     atom    1 type  1   force =     0.00000000    0.00000000   -0.01049299
     atom    2 type  1   force =     0.00000000    0.00000000    0.01049299
```
```console
          total   stress  (Ry/bohr**3)                   (kbar)     P=       -4.00
  -0.00003190  -0.00000000   0.00000000           -4.69       -0.00        0.00
  -0.00000000  -0.00003190   0.00000000           -0.00       -4.69        0.00
   0.00000000   0.00000000  -0.00001767            0.00        0.00       -2.60
```

are reduced to

```console
     Forces acting on atoms (cartesian axes, Ry/au):

     atom    1 type  1   force =     0.00000000    0.00000000    0.00047838
     atom    2 type  1   force =     0.00000000    0.00000000   -0.00047838
```
```console
          total   stress  (Ry/bohr**3)                   (kbar)     P=        0.07
   0.00000070   0.00000000   0.00000000            0.10        0.00        0.00
   0.00000000   0.00000070   0.00000000            0.00        0.10        0.00
   0.00000000   0.00000000   0.00000006            0.00        0.00        0.01
```

Input/output cell:

```python
In [1]: relax_wc = load_node(16355)

In [2]: relax_wc.inputs.structure.cell
Out[2]: 
[[3.6086769266, 0.0, 0.0],
 [-1.8043384633, 3.1252058924, 0.0],
 [0.0, 0.0, 21.3114930844]]

In [3]: relax_wc.outputs.output_structure.cell
Out[3]: 
[[3.5615977430325, 0.0, 0.0],
 [-1.7807988715162, 3.0844341234181, 0.0],
 [0.0, 0.0, 21.311492898049]]
```